hack/generator/azure-arm.yaml

```diff
@@ -2,9 +2,9 @@ schemaUrl: 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTem
 schemaUrlRewrite:
   from: 'https://schema.management.azure.com/schemas/'
   to: 'specs/azure-resource-manager-schemas/schemas/'
-destinationGoModuleFile: ../generated/go.mod
+destinationGoModuleFile: ../../v2/go.mod
 typesOutputPath: ../../v2/api
```

Comment:
one change I made when sketching this was to have the `typesOutputPath` be relative to the destination module file (since `typeRegistrationOutputFile` already is), but I can make this change later